### PR TITLE
LPD-36199 Add and enable hook to use keyboard movement

### PR DESCRIPTION
**Pull request by sandrodw3**
:information_source:  Only **LPD-36199** commits need to be reviewed

### CODE REVIEW

**Comment by liferay-page-management** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by sandrodw3** [comment]
ci:test:page-management-playwright

File: modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/miller_columns/hooks/useKeyboardNavigation.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+		isTarget,
+	};
+}
+
+function getNextTarget({
+	columnSizes,
+	item: {active, columnIndex, hasChild, itemIndex, parentIndex},
+	key,
+}: {
+	columnSizes: number[];
```
Comment (georgel-pop-lr) [inline modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/miller_columns/hooks/useKeyboardNavigation.ts:134]: Maybe we can rename this to columnLengths, for clarity.

**Comment by sandrodw3** [comment]
ci:test:page-management

**Comment by sandrodw3** [comment]
ci:test:page-management

File: modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/miller_columns/MillerColumnsItem.js [hunk 2]
```diff
@@ -410,7 +410,21 @@ const MillerColumnsItem = ({
 			ref={ref}
 			verticalAlign="center"
 		>
-			<a className="miller-columns-item-mask" href={url} role="button">
+			<a
+				className="miller-columns-item-mask"
+				href={url}
+				onKeyDown={(event) => {
+					if (
+						event.key === 'ArrowRight' &&
```
Comment (victorg1991) [inline modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/miller_columns/MillerColumnsItem.js:411]: We probably need to take into account RTL here

File: modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/css/miller_columns/_MillerColumns.scss [hunk 3]
```diff
@@ -96,11 +96,18 @@
 		z-index: 1;
 
 		&:focus {
-			box-shadow: inset 0 0 0 0.125rem $primary-l1;
+			box-shadow: none;
 			outline: 0;
 		}
 	}
 
+	&:not(.dragging):not(.drop-bottom):not(.drop-middle):not(.drop-top)
```
Comment (victorg1991) [inline modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/css/miller_columns/_MillerColumns.scss:104]: not, not, not, not 😢

Comment (victorg1991) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/css/miller_columns/_MillerColumns.scss:104]: it found it funny

Comment (sandrodw3) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/css/miller_columns/_MillerColumns.scss:104]: 🤣

File: modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/miller_columns/MillerColumnsItem.js (hunk 2 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/miller_columns/MillerColumnsItem.js:411]: True

Comment (sandrodw3) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/miller_columns/MillerColumnsItem.js:411]: Done! I applied also rtl to the hooks, I completely forgot that, thanks!

**Comment by sandrodw3** [comment]
ci:test:page-management

**Comment by sandrodw3** [comment]
ci:test:page-management

**Comment by sandrodw3** [comment]
ci:forward

**Comment by ealonso** [comment]
see: https://github.com/brianchandotcom/liferay-portal/pull/156315

**Pull request closed without merging**